consul/src/main/java/com/linecorp/armeria/server/consul/ConsulUpdatingListenerBuilder.java

```diff
@@ -0,0 +1,188 @@
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.net.URI;
+import java.time.Duration;
+
+import javax.annotation.Nullable;
+
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.SessionProtocol;
+import com.linecorp.armeria.internal.consul.ConsulClientBuilder;
+import com.linecorp.armeria.server.Server;
+
+/**
+ * Builds a new {@link ConsulUpdatingListener}, which registers the server to Consul cluster.
+ * <h3>Examples</h3>
+ * <pre>{@code
+ * ConsulUpdatingListener listener = ConsulUpdatingListener.builder("myService")
+ *                                                         .consulPort(8501")
+ *                                                         .build();
+ * server.addListener(listener);
+ * }</pre>
+ */
+public final class ConsulUpdatingListenerBuilder extends ConsulClientBuilder {
+
+    private static final String DEFAULT_CHECK_INTERVAL = "10s";
+    private final String serviceName;
+
+    @Nullable
+    private Endpoint serviceEndpoint;
+    @Nullable
+    private URI checkUri;
+    @Nullable
+    private String checkInterval;
+    @Nullable
+    private HttpMethod checkMethod;
+
+    /**
+     * Creates a {@link ConsulUpdatingListenerBuilder} with a service name.
+     *
+     * @param serviceName the service name to register
+     */
+    ConsulUpdatingListenerBuilder(String serviceName) {
+        this.serviceName = requireNonNull(serviceName, "serviceName");
+        checkArgument(!this.serviceName.isEmpty(), "serviceName can't be empty");
+    }
+
+    @Override
+    public ConsulUpdatingListenerBuilder consulUri(URI consulUri) {
+        return (ConsulUpdatingListenerBuilder) super.consulUri(consulUri);
+    }
+
+    @Override
+    public ConsulUpdatingListenerBuilder consulUri(String consulUri) {
+        return (ConsulUpdatingListenerBuilder) super.consulUri(consulUri);
+    }
+
+    @Override
+    public ConsulUpdatingListenerBuilder consulProtocol(SessionProtocol consulProtocol) {
+        return (ConsulUpdatingListenerBuilder) super.consulProtocol(consulProtocol);
+    }
+
+    @Override
+    public ConsulUpdatingListenerBuilder consulAddress(String consulAddress) {
+        return (ConsulUpdatingListenerBuilder) super.consulAddress(consulAddress);
+    }
+
+    @Override
+    public ConsulUpdatingListenerBuilder consulPort(int consulPort) {
+        return (ConsulUpdatingListenerBuilder) super.consulPort(consulPort);
+    }
+
+    @Override
+    public ConsulUpdatingListenerBuilder consulApiVersion(String consulApiVersion) {
+        return (ConsulUpdatingListenerBuilder) super.consulApiVersion(consulApiVersion);
+    }
+
+    @Override
+    public ConsulUpdatingListenerBuilder consulToken(String consulToken) {
+        return (ConsulUpdatingListenerBuilder) super.consulToken(consulToken);
+    }
```

Comment:
Could you move these methods overridden only return type to the bottom of this class?

Reply:
Ok.